pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,68 @@
+package agent
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	// Cluster name
+	Name string `json:"name"`
+	// Installation will start once cluster is in known state and approved
+	Approved                 bool   `json:"approved,omitempty"`
+	OpenshiftVersion         string `json:"openshiftVersion"`
+	BaseDnsDomain            string `json:"baseDnsDomain,omitempty"`
+	ClusterNetworkCidr       string `json:"clusterNetworkCidr,omitempty"`
+	ClusterNetworkHostPrefix int64  `json:"clusterNetworkHostPrefix,omitempty"`
+	ServiceNetworkCidr       string `json:"serviceNetworkCidr,omitempty"`
+	ApiVip                   string `json:"apiVip,omitempty"`
+	ApiVipDnsName            string `json:"apiVipDnsName,omitempty"`
+	IngressVip               string `json:"ingresVip,omitempty"`
+	MachineNetworkCidr       string `json:"machineNetworkCidr,omitempty"`
+	SshPublicKey             string `json:"sshPublicKey,omitempty"`
```

Comment:
Should be SSHPublicKey.

We have CD.Spec.Provisioning.SSHPrivateKeySecretRef. Could we re-use? If we've got the private key, we've got the public key.

Reply:
Assisted service does it's own log gathering if failed, doesn't require a private key. 

Sounds like we do want to expose just a public ssh key in this scenario. Likely should move higher up in Hive cluster deployment. We do not want to force to give a private key.